ginga/trcalc.py

```diff
@@ -965,3 +965,61 @@ def get_minmax(dtype):
         info = np.finfo(dtype)
 
     return info.min, info.max
+
+
+def fancy_index(d_obj, view):
+    """Return a slice from a data object according to a view.
+
+    Parameters
+    ----------
+    d_obj : numpy ndarray, dask array or zarr array
+        2D or 3D data array
+
+    view : tuple of slice or int array
+        View into the array
+
+    Returns
+    -------
+    arr : ndarray
+        The result of the fancy index as a numpy array
+
+    d_obj can define a 2D or 3D array which can be a numpy, dask or zarr
+    array.  We always return a numpy array.
+    """
+    if not isinstance(view[0], slice):
+        # <-- indicates fancy indexing being used instead of slices
+
+        if isinstance(d_obj, np.ndarray):
+            # <-- numpy array
+            view = np.ix_(*view)
+
+        # duck-typing test for zarr object
+        elif hasattr(d_obj, 'get_coordinate_selection'):
+            # <-- zarr object
+            # zarr does not support numpy-style fancy indexing for 2D
+            # and higher arrays, so we need to use the
+            # get_coordinate_selection() method to fetch all the values
+            shape = [len(idxs) for idxs in view]
+            _m = np.meshgrid(*tuple(reversed(view)))
+            iarrs = [_m[i].reshape(-1) for i in range(len(_m))]
+            arr = d_obj.get_coordinate_selection(tuple(reversed(iarrs)))
+            arr = arr.reshape(shape)
+            return arr
+
+        else:
```

Comment:
Can you guarantee that `d_obj` would definitely be Dask object at this point? Or should this be an `elif` instead?

Reply:
good question. The current code sort of assumes that by process of elimination.  Probably it should be an `elif` with an exception raised if it didn't match anything before.  Problem is that I want to detect the cases without having to import `zarr` or `dask`, because this becomes *the* basic slicing function.  So if a good duck-typing test could be done that might be a possibility...

Reply:
At this point I think the code for this is ok, we can refactor it later if a better test for dask arrays can be found.